classes/package.php

```diff
@@ -262,33 +268,33 @@ public function delete_from_db() {
      * Get a specific package by its hash from the db.
      *
      * @param string $hash
-     * @return package
-     * @throws \dml_exception
+     * @return ?array [id of database record, package instance] or null if not found
+     * @throws dml_exception
+     * @throws moodle_exception
      */
-    public static function get_record_by_hash(string $hash): package {
+    public static function get_record_by_hash(string $hash): ?array {
         global $DB;
-        $package = (array)$DB->get_record('qtype_questionpy_package', ['hash' => $hash]);
-        list($language, $name, $description) = self::get_languagedata($package["id"]);
-        $tags = self::get_tagdata($package["id"]);
-        $temp = [
-            'languages' => $language,
-            'name' => $name,
-            'description' => $description,
-            'tags' => $tags
-        ];
-        $package = array_merge($package, $temp);
-        return array_converter::from_array(self::class, $package);
+        $package = $DB->get_record('qtype_questionpy_package', ['hash' => $hash]);
+        if (!$package) {
+            return null;
+        }
+
+        list($package->languages, $package->name, $package->description) = self::get_languagedata($package->id);
```

Comment:
```suggestion
        [$package->languages, $package->name, $package->description] = self::get_languagedata($package->id);
```
Würde ich ansprechender finden (da es Python ähnelt😄 ).

Reply:
Ich auch! Kannte ich bisher nicht (scheint auch in der Doku zu fehlen :thinking:)